.github/workflows/go-getter.yml

```
@@ -0,0 +1,156 @@
+name: go-getter
+
+on: [push]
+
+env:
+  TEST_RESULTS_PATH: /tmp/test-results
+
+jobs:
+
+  linux-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version:
+          - 1.19
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Create test directory
+        run: |
+          mkdir -p ${{ env.TEST_RESULTS_PATH }}
+
+      - name: Download go modules
+        run: go mod download
+      
+      - name: Cache / restore go modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      
+      # Check go fmt output because it does not report non-zero when there are fmt changes
+      - name: Run gofmt
+        run: |
+          go fmt ./...
+          files=$(go fmt ./...)
+            if [ -n "$files" ]; then
+              echo "The following file(s) do not conform to go fmt:"
+              echo "$files"
+              exit 1
+            fi
+      
+      - name: Install gotestsum
+        run: go install gotest.tools/gotestsum@v1.8.2
+
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::388664967494:role/hc-go-getter-test
+          role-session-name: ${{ github.run_id }}
+          audience: https://github.com/hashicorp
+
+      - name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0.4.0'
+        with:
+          workload_identity_provider: 'projects/328212837253/locations/global/workloadIdentityPools/hc-go-getter-test/providers/hc-go-getter-test'
+          service_account: hc-go-getter-test@hc-e56c0f7c21c448d2be9e7696073.iam.gserviceaccount.com
+          audience: https://github.com/hashicorp
+        
+      - name: Run go tests
+        run: |
+          PACKAGE_NAMES=$(go list ./...)
+          echo "Running $(echo $PACKAGE_NAMES | wc -w) packages"
+          echo $PACKAGE_NAMES
+          gotestsum --format=short-verbose --junitfile $TEST_RESULTS_PATH/go-getter/gotestsum-report.xml -- -p 2 -cover -coverprofile=linux_cov.part $PACKAGE_NAMES
+      
+      # Save coverage report parts
+      - name: Upload and save artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: linux test results
+          path: linux_cov.part
+
+  windows-tests:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        go-version:
+          - 1.19
```

Similar to my suggestion in the Linux test steps, I suspect we'd be able to test `1.17`, `1.18`, and `1.19`?

```suggestion
        go-version:
          - 1.17
          - 1.18
          - 1.19
```